### ISSUE TITLE
German world campaign issue Narvik map

### ISSUE THREAD

**Issue by MansteinlovesGuns** [nicupavel/openpanzer#226]
Hello

Second time playing this campaign as I had too many  brilliant victories and missed the majority of maps because of this.

Second time around I got to Narvik the Norway map.  On turn 2 the computer  starts their turn but then fails to complete it. It does not freeze as I can move around and click on units to see their Names.

I tried reloading to my previous turn multiple times and believe I found the issue:

When the computer turn 2 plays their aircraft carrier it moves into position but then there is the sound of a unit dying. This is when the computer makes no further moves to finish its turn and I am unable to take my next turn 3.

I am on android. Playing through your github link for beta

**Comment by salexgit** [comment]
Hi

Could you please upload the saved game so that we can try to reproduce the problem?

**Comment by MansteinlovesGuns** [comment]
I accidentally saved over that cloud save seconds after your email came in.
Played on my smartphone.


The computer played fine its first turn and started off okay it moves its
aircraft carrier then nothing further happens in their turn after the
carrier makes the unit loss sound bite scream.  The message the computer is
playing stays but nothing happens. I can move around the screen, but the
computer turn never ends and I left it after a few minutes.


Narvik map

**Comment by salexgit** [comment]
Hi

If possible, can you replay the game and provide a saved file illustrating the problem?

Regards, Alex

**Comment by MansteinlovesGuns** [comment]
I will do so.  How do I upload to github fr my phone? Send you the cloud
save link?

**Comment by salexgit** [comment]
Hi

You can either post here the cloud save ID or you can compress and attach the actual saved game. The latter option is a bit better because the cloud save can be overridden. The only tricky part is that if you play with phone, then it could be difficult to find where exactly the saved game is. See discussion #205 for further information...

Regards, Alex

**Comment by MansteinlovesGuns** [comment]
http://panzermarshal.com/beta/

Hopefully this works if it doesnt I have the save file.  Same issue occured when I replayed Narvik. Computer turn two as soon the the carrier shows up the issue happens

**Comment by MansteinlovesGuns** [comment]
http://panzermarshal.com/beta/

There is the narvik save.  If you end the turn you will see the issue occur
when the aircraft carrier arrives

**Comment by Themaginotlineww2** [comment]
I am having the same problem

**Comment by salexgit** [comment]
Hi all

I played Narvik scenario with online beta and I did not have any issue. Can you please confirm that this problem occurred only on the mobile phone? 
It could be the case that the problem is not with the scenario per se, but rather with the mobile platform (for which there are other issues as well).

**Comment by MansteinlovesGuns** [comment]
I played on my mobile android.  I only play the mobile app on android

**Comment by salexgit** [comment]
Hi

To isolate problem, can you please play the same scenario in a same way on the desktop PC by using online beta? If you do not encounter any problem, then the issue is due to the mobile platform.
By the way, this is also how you can circumvent around the problem: play this particular scenario on the desktop PC and then continue using mobile version.

**Comment by Themaginotlineww2** [comment]
Im playing on a mac and I had the same problem. It does seem to happen only after a aircraft carrier comes into view.

**Comment by salexgit** [comment]
Hi

I can try re-play the scenario on MAC and see whether I will be able to reproduce it.

**Comment by MansteinlovesGuns** [comment]
Playing US world campaign on my phone. Same issue with game stopping. This
time on leningrad map.  Turn two a JU 52 attacked my tank and stopped the
game.  Same unit death sound as before with Narvik

**Comment by MansteinlovesGuns** [comment]
Hey dude.

I was replaying this campaign on my phone to play the levels I have to yet
to play.

When I came back to Eindhoven I few issues came up that were new.

Pill boxes I had destroyed were removed from map but in their hex they
still shot at my units.  I also noticed non destroyed pill boxes shooting
at multiple enemies.

A unit I destroyed also shot at me as well as an invisible hex. This is all
new as I have played the Eindhoven map at least 6 times.

**Comment by jonh64** [comment]
I've had the same issue in 'Narvik' and again later with 'Malta' when the aircraft carrier appears.  After several attempts i managed to lure the carrier out and kill it before it could fire.
JU52 issue reappears in the US world campaign 'Leningrad' scenario when game freezes when they both attack.
Another issue is with the US world campaign 'D-Day' scenario when none of my air fighter units could move after turn 1.

**Comment by MansteinlovesGuns** [comment]
I was able to get around the carrier as well by keeping it away from my
planes.  In Afrika corps American paraxhute planes appeared and stalled the
game as well.

I also had the carrier issue happen in the US campaign.

As well in Sevastopol if you place units in a landing crafr and the are
fired on by ememy landing crafts the same issue caused by the carrier
occurs.

**Comment by nicupavel** [comment]
Should be fixed on 3.2.6.

**Comment by jonh64** [comment]
hopefully it is in 3.26 because it's still there on 3.25
